### PR TITLE
[MM-20804] Add "Unarchive Channel" option to the channel info screen - Additional fixes

### PR DESCRIPTION
**Pull request by allenlai18**
#### Summary

These are small fixes needed to the PR.
https://github.com/mattermost/mattermost-server/pull/13349

1. the username needs to be passed to the Post Props because of how system_messages are being rendered in the webapp and mobile.

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -632,6 +632,9 @@ func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Chan
 			Message:   T("api.channel.restore_channel.unarchived", map[string]interface{}{"Username": user.Username}),
 			Type:      model.POST_CHANNEL_RESTORED,
 			UserId:    userId,
+			Props: model.StringInterface{
+				"username": user.Username,
```
Comment (hanzei) [inline app/channel.go]: I'm going to deffer the decision about this towards @jespino, but this looks bad to me. The username can be updated, and hence the data in `Props` might be outdated.

Comment (jespino) [reply app/channel.go]: We can investigate it a bit more, probably passing here the userId (here and in all the other places), but for now is consistent with the rest of the implementation. (See for example you have the same problem in `POST_CHANNEL_DELETED` post). My proposal is to create another ticket to handle the specific problem of username changes on this kind of messages, but I don't see including that change as part of this issue.

Comment (jespino) [reply app/channel.go]: Added the ticket here: https://mattermost.atlassian.net/browse/MM-21711

**Comment by allenlai18** [comment]
@jasonblais   can you assign this to another QA please?

**Comment by jasonblais** [comment]
Thanks for the heads up Allen - @josephbaylon assigning to you since you're looking at a related mobile PR. Can you help review tomorrow if you have the time?

**Comment by josephbaylon** [comment]
@jasonblais Sure I'll take a look after my morning meetings.

**Comment by allenlai18** [comment]
@jespino    you were right. updating the channel object's delete_at = 0 .  fixed the issue.

**Comment by jespino** [comment]
Awesome! :)

**Comment by allenlai18** [comment]
> @allenlai18 I'm testing this server in conjunction with a mobile build from this PR [mattermost/mattermost-mobile#3663](https://github.com/mattermost/mattermost-mobile/pull/3663) as I believe the webapp fix is not available yet.

@josephbaylon 

it should be in the channel info. that is really weird, are you sure its the right branch?  you also have to use the redux PR.

**Comment by josephbaylon** [comment]
> > @allenlai18 I'm testing this server in conjunction with a mobile build from this PR [mattermost/mattermost-mobile#3663](https://github.com/mattermost/mattermost-mobile/pull/3663) as I believe the webapp fix is not available yet.
> 
> @josephbaylon
> 
> it should be in the channel info. that is really weird, are you sure its the right branch? you also have to use the redux PR.



> > @allenlai18 I'm testing this server in conjunction with a mobile build from this PR [mattermost/mattermost-mobile#3663](https://github.com/mattermost/mattermost-mobile/pull/3663) as I believe the webapp fix is not available yet.
> 
> @josephbaylon
> 
> it should be in the channel info. that is really weird, are you sure its the right branch? you also have to use the redux PR.
@allenlai18 I used the following for testing,

This is the generated APK for your mobile PR -->
https://pr-builds.mattermost.com/mattermost-mobile/build-pr-3663/Mattermost_Beta.apk

Here's the mattermost server in this PR -->
 https://mattermost-server-pr-13614.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

Regarding redux, I'm not sure how to tie that into the testing and/or mm server. Please provide full steps on how to test this PR.

**Comment by allenlai18** [comment]
> @allenlai18 I used the following for testing,
> 
> This is the generated APK for your mobile PR -->
> https://pr-builds.mattermost.com/mattermost-mobile/build-pr-3663/Mattermost_Beta.apk
> 
> Here's the mattermost server in this PR -->
> https://mattermost-server-pr-13614.test.mattermost.cloud
> 
> Account Type	Username	Password
> Admin	sysadmin	Sys@dmin123
> User	user-1	User-1@123
> Regarding redux, I'm not sure how to tie that into the testing and/or mm server. Please provide full steps on how to test this PR.

@josephbaylon  for web, we need to use the webapp PR https://github.com/mattermost/mattermost-webapp/pull/4458

**Comment by josephbaylon** [comment]
> @josephbaylon for web, we need to use the webapp PR [mattermost/mattermost-webapp#4458](https://github.com/mattermost/mattermost-webapp/pull/4458)
> also, how do i run a .apk file?

Hi @allenlai18 , per @jasonblais (https://github.com/mattermost/mattermost-webapp/pull/4458), server needs to be merged first before we can test the webapp.

For running the apk, just download and install in your device (either emulator or physical).

**Comment by jasonblais** [comment]
@allenlai18 Now that the redux PR is merged, I'm wondering if we should update the redux hash in the mobile PR https://github.com/mattermost/mattermost-mobile/pull/3663 to see if that resolves the issue?

@josephbaylon this server PR shouldn't effect the visibility of the "Unarchive Channel" option on mobile, so I wonder if the .apk file for the mobile PR is capturing the changes from https://github.com/mattermost/mattermost-mobile/pull/3663 (and if it is, perhaps it's missing those redux changes)

@allenlai18 
> also, how do i run a .apk file?

You can click on the link below from mobile browser - this will download a Mattermost Beta app on your phone with the changes from https://github.com/mattermost/mattermost-mobile/pull/3663.

https://pr-builds.mattermost.com/mattermost-mobile/build-pr-3663/Mattermost_Beta.apk

Note that if you already had Mattermost Beta app installed, you need to delete it before downloading the above .apk file.

Let me know if that's helpful?

**Comment by allenlai18** [comment]
> @allenlai18 Now that the redux PR is merged, I'm wondering if we should update the redux hash in the mobile PR [mattermost/mattermost-mobile#3663](https://github.com/mattermost/mattermost-mobile/pull/3663) to see if that resolves the issue?
> 
> @josephbaylon this server PR shouldn't effect the visibility of the "Unarchive Channel" option on mobile, so I wonder if the .apk file for the mobile PR is capturing the changes from [mattermost/mattermost-mobile#3663](https://github.com/mattermost/mattermost-mobile/pull/3663) (and if it is, perhaps it's missing those redux changes)
> 
> @allenlai18
> 
> > also, how do i run a .apk file?
> 
> You can click on the link below from mobile browser - this will download a Mattermost Beta app on your phone with the changes from [mattermost/mattermost-mobile#3663](https://github.com/mattermost/mattermost-mobile/pull/3663).
> 
> https://pr-builds.mattermost.com/mattermost-mobile/build-pr-3663/Mattermost_Beta.apk
> 
> Note that if you already had Mattermost Beta app installed, you need to delete it before downloading the above .apk file.
> 
> Let me know if that's helpful?

@jasonblais i do not have an android phone

**Comment by allenlai18** [comment]
> @allenlai18 @jasonblais
> See comment [mattermost/mattermost-mobile#3663 (review)](https://github.com/mattermost/mattermost-mobile/pull/3663#pullrequestreview-346777231)
> 
> Also, I tried twice to spin a new test server but failed so I am not able to proceed with retesting. Looks like this PR needs sync and merge according to errors below.

@josephbaylon  i think for mm staff, you can run ```/update-branch``` git slash command to fix the sync and merge problem if there are no conflicts

**Comment by josephbaylon** [comment]
@allenlai18 Looks like an issue with PR merge

```
+ cd /go/src/github.com/mattermost/mattermost-server

+ echo Installing golangci-lint

Installing golangci-lint

+ curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh

+ sh -s -- -b /usr/local/bin v1.23.0

golangci/golangci-lint info checking GitHub for tag 'v1.23.0'

golangci/golangci-lint info found version: 1.23.0 for v1.23.0/linux/amd64

golangci/golangci-lint info installed /usr/local/bin/golangci-lint

+ echo Installing mattermost-govet

Installing mattermost-govet

+ GO111MODULE=off go get -u github.com/mattermost/mattermost-govet

+ export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/go/bin:/go/src/github.com/mattermost/mattermost-server/bin

+ make config-reset

Resetting configuration to default

rm -f config/config.json

OUTPUT_CONFIG=/go/src/github.com/mattermost/mattermost-server/config/config.json go generate  -mod=vendor ./config

+ make check-style BUILD_NUMBER=PR-13614-7 MM_NO_ENTERPRISE_LINT=true

Running golangci-lint

golangci-lint run ./...

level=warning msg="[runner] Can't process result by autogenerated_exclude processor: can't filter issue result.Issue{FromLinter:\"golint\", Text:\"exported var SystemEmojis should have comment or be unexported\", SourceLines:[]string(nil), Replacement:(*result.Replacement)(nil), Pkg:(*packages.Package)(0xc001800000), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:\"model/emoji_data.go\", Offset:132, Line:6, Column:5}, HunkPos:0}: failed to get doc of file model/emoji_data.go: failed to scan file: bufio.Scanner: token too long"

imports/imports.go:7:2: a blank import should be only in a main or test package, or have a comment justifying it (golint)

	_ "github.com/mattermost/mattermost-server/v5/enterprise/account_migration"

	^

make: *** [Makefile:166: golangci-lint] Error 1

script returned exit code 2
```

**Comment by allenlai18** [comment]
@josephbaylon     @jespino  its a known issue where the build fails right?

**Comment by jasonblais** [comment]
@metanerd Can you help with the build failure noted above? Might be something in our pipeline.

image

**Comment by jasonblais** [comment]
cc @josephbaylon on above

**Pull request merged**